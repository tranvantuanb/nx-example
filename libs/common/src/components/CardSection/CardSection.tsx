import React from 'react';
import tw, { styled } from 'twin.macro';

import { Text } from '../Text/Text';

export interface CardSectionProps {
  className?: string;
  header?: React.ReactNode;
  children?: React.ReactNode;
  noMargin?: boolean;
}

const StyledCardSection = styled.div<Pick<CardSectionProps, 'noMargin'>>`
  ${({ noMargin }) => !noMargin && tw`mb-[32px]`}
`;

export const CardSection: React.FC<CardSectionProps> = React.memo(
  ({ className, header, noMargin, children, ...props }) => {
    return (
      <StyledCardSection noMargin={noMargin} className={className} {...props}>
        {!!header && (
          <Text.Caption size="sm" uppercase className="mb-[16px]">
            {header}
          </Text.Caption>
        )}
        {children}
      </StyledCardSection>
    );
  }
);
