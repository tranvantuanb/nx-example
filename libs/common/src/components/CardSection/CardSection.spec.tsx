import { render } from '../../test';

import { CardSection, CardSectionProps } from './CardSection';

describe('CardSection', () => {
  it('should render successfully', () => {
    const defaultProps: CardSectionProps = {
      header: 'Section',
    };
    const { container } = render(<CardSection {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
