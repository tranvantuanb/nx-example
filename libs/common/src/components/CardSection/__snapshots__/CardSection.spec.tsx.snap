// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSection should render successfully 1`] = `
.c0 {
  display: block;
}

.c1 {
  margin-bottom: 16px;
}

<div>
  <div
    class="css-1ijzjjj-StyledCardSection e5y6tk60"
  >
    <span
      class="ant-typography c0 c1 css-14xdwh8-StyledText-StyledTextCaption emffahp0"
    >
      <strong>
        Section
      </strong>
    </span>
  </div>
</div>
`;
